### PR TITLE
[ACA-4258] Add Task Filter Counters

### PR DESCRIPTION
**Pull request by davidcanonieto**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/ACA-4258

**What is the new behaviour?**
This PR brings filter counters so that the user can see at a glance how many tasks are inside each filter.
task-filter-counter

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/task/task-filters/components/base-task-filters-cloud.component.html [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
           [attr.aria-label]="filter.name | translate"
           [id]="filter.id"
           [attr.data-automation-id]="filter.key + '_filter'"
           mat-button
+          [class.adf-active]="currentFilter === filter"
           class="adf-filter-action-button adf-full-width" fxLayout="row" fxLayoutAlign="space-between center">
           <adf-icon data-automation-id="adf-filter-icon" *ngIf="showIcons" [value]="filter.icon"></adf-icon>
           <span data-automation-id="adf-filter-label" class="adf-filter-action-button__label">{{ filter.name | translate }}</span>
         </button>
+        <span *ngIf="counters$[filter.key]" class="adf-filter-action-button__counter">
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/task/task-filters/components/base-task-filters-cloud.component.html:13]: wouldn't be better to have a counter as directive or pipe on its own and be able to consume it in different places of the app?

Comment (mauriziovitale) [reply lib/process-services-cloud/src/lib/task/task-filters/components/base-task-filters-cloud.component.html:13]: something on top of https://material.angular.io/components/badge/overview

**Review by eromano — APPROVED** [review]

File: lib/process-services-cloud/src/lib/task/task-filters/services/task-filter-cloud.service.ts [hunk 2]
```diff
@@ -219,6 +223,27 @@ export class TaskFilterCloudService {
         return defaultFilters.findIndex((filter) => filterName === filter.name) !== -1;
     }
 
+    /**
+     * Finds a task using an object with optional query properties.
+     * @param requestNode Query object
+     * @returns Task information
+     */
+    getTaskFilterCounter(taskFilter: TaskFilterCloudModel): Observable<any> {
+        if (taskFilter.appName || taskFilter.appName === '') {
```
[the remark this answers is not in the document]
Comment (davidcanonieto) [reply lib/process-services-cloud/src/lib/task/task-filters/services/task-filter-cloud.service.ts:232]: This widget is also available in community that is why we do this check all over the process cloud repo. If we want to keep it enterprise then we need to come up with a different solution.

File: lib/process-services-cloud/src/lib/task/task-filters/components/base-task-filters-cloud.component.html (hunk 1 above)
Comment (davidcanonieto) [reply lib/process-services-cloud/src/lib/task/task-filters/components/base-task-filters-cloud.component.html:13]: I agree on the directive, however we're limited in the design as this feature follows to the teeth the designs created by the UI team

**Pull request merged**